lib/mongo/uri.rb

```diff
@@ -405,27 +405,23 @@ def parse_uri_options!(string)
 
     def parse_user!(string)
       if (string && user = string.partition(AUTH_USER_PWD_DELIM)[0])
-        if user.length > 0
```

Comment:
Removed this line to allow for a distinction between a URI like `"mongodb://localhost/"` (valid) and `"mongodb://@localhost.com/"` (invalid).